plugins/policy/processor/match_label_selector.go

```diff
@@ -0,0 +1,329 @@
+			!pp.isNsUpdateExpressionMatch(ns, matchNsUpdateExpressions) {
+			return false
+		}
+		return true
+
+	} else if len(matchNsUpdateLabels) == 0 && len(matchNsUpdateExpressions) > 0 {
+		if !pp.isNsUpdateExpressionMatch(ns, matchNsUpdateExpressions) {
+			return false
+		}
+		return true
+
+	} else if len(matchNsUpdateLabels) > 0 && len(matchNsUpdateExpressions) == 0 {
+		if !pp.isNsUpdateLabelMatch(ns, matchNsUpdateLabels) {
+			return false
+		}
+		return true
+
+	}
+
+	// Empty label selector (both podLabels and podExpressions)
+	// returns true as it matches all the resources
+	return true
+}
+
+// isNsUpdateLabelMatch finds if namespace LabelSelector matches Pod's namespace labels,
+// after an update in namespace labels
+func (pp *PolicyProcessor) isNsUpdateLabelMatch(ns *nsmodel.Namespace,
+	matchNsUpdateLabels []*policymodel.Policy_Label) bool {
+	//Get Pod's namespace labels
+	_, nsData := pp.Cache.LookupNamespace(nsmodel.GetID(ns))
+	nsLabels := nsData.Label
+
+	// create a map with current namespace's labels
+	labelExists := make(map[string]bool)
+	labelPrefix := ""
+	for _, nsLabel := range nsLabels {
+		labelExistsKey := nsLabel.Key + "/" + nsLabel.Value
+		labelExists[labelExistsKey] = true
+	}
+
+	return isMatchLabel(matchNsUpdateLabels, labelExists, labelPrefix)
+}
+
+// isNsUpdateExpressionMatch finds if pod's namespace labels match a collection of namespace selector expressions,
+// after an update in namespace labels
+func (pp *PolicyProcessor) isNsUpdateExpressionMatch(ns *nsmodel.Namespace,
+	matchNsUpdateExpressions []*policymodel.Policy_LabelSelector_LabelExpression) bool {
+
+	//Get Pod's namespace labels
+	_, nsData := pp.Cache.LookupNamespace(nsmodel.GetID(ns))
+	nsLabels := nsData.Label
+
+	// create a map with current namespace's labels
+	labelExists := make(map[string][]string)
+	labelPrefix := ""
+	for _, nsLabel := range nsLabels {
+		labelExistsKey := nsLabel.Key
+		labelExists[labelExistsKey] = append(labelExists[labelExistsKey], nsLabel.Value)
+	}
+
+	return isMatchExpression(matchNsUpdateExpressions, labelExists, labelPrefix)
+}
+
+// isMatchLabel checks if labels are matching a given map of existing labels that we compare against
+func isMatchLabel(matchLabels []*policymodel.Policy_Label, labelExists map[string]bool, matchLabelPrefix string) bool {
+
+	isMatch := true
+	for _, matchLabel := range matchLabels {
+		labelExistsKey := matchLabelPrefix + matchLabel.Key + "/" + matchLabel.Value
+		if labelExists[labelExistsKey] != true {
+			isMatch = false
+			break
+		}
+	}
+
+	return isMatch
+}
+
+// isMatchExpression checks if expressions are matching a given map of existing expressions that we compare against
+func isMatchExpression(matchExpressions []*policymodel.Policy_LabelSelector_LabelExpression,
+	keyMap map[string][]string, expressionKeyPrefix string) bool {
+
+	isMatch := false
+	for _, matchExpression := range matchExpressions {
+		switch matchExpression.Operator {
+		case in:
```

Comment:
Please double-check implementation of operators.
- keyMap is checked with "prefix+key+value" as key, while callers fill it with keys only
- isMatch gets overwritten after each iteration of "for _, label := range labels"
- the inner for loop is probably unecessary, there is always one value assigned to a key at most 
- doesNotExist returns false when "len(keyMap[expressionKey]) == 0" ???